extensions-core/reactive-executor/deployment/src/main/java/org/apache/camel/quarkus/reactive/executor/deployment/BuildProcessor.java

```diff
@@ -25,9 +25,16 @@
 import org.apache.camel.quarkus.reactive.executor.ReactiveExecutorRecorder;
 
 public class BuildProcessor {
+    @Record(value = ExecutionTime.STATIC_INIT, optional = true)
+    @BuildStep(onlyIf = Flags.MainEnabled.class)
+    CamelReactiveExecutorBuildItem reactiveExecutor(ReactiveExecutorRecorder recorder) {
+        return new CamelReactiveExecutorBuildItem(recorder.createReactiveExecutor());
+    }
+
     @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
     @BuildStep(onlyIf = Flags.MainEnabled.class)
-    CamelReactiveExecutorBuildItem reactiveExecutor(ReactiveExecutorRecorder recorder, VertxBuildItem vertx) {
-        return new CamelReactiveExecutorBuildItem(recorder.createReactiveExecutor(vertx.getVertx()));
+    void initReactiveExecutor(ReactiveExecutorRecorder recorder, CamelReactiveExecutorBuildItem executor,
+            VertxBuildItem vertx) {
```

Comment:
I think we need a sort of barrier to avoid this to be executed after main is started

Reply:
Right, I'll add a BuildItem to ensure the start of the context is done after.

Reply:
We may eventually use the `Consume/Produce` annotations with an [empty build item ](https://quarkus.io/guides/writing-extensions#empty-build-items)

Reply:
Well, I've just added an `CamelInitializedReactiveExecutorBuildItem` to ensure a 2 phase setup for the reactive executor.